hrm-domain/hrm-core/permissions/actions.ts

```diff
@@ -103,202 +101,3 @@ export const isValidSetOfActionsForTarget = <T extends TargetKind>(
     actions.every(action => validActionsOnTarget.includes(action))
   );
 };
-
-// deep-diff lib kinds:
-const NEW_PROPERTY = 'N';
-const EDITED_PROPERTY = 'E';
-const ARRAY_CHANGED = 'A';
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const DELETED_PROPERTY = 'D';
-
-/**
- * Compares the target.length first components of a change path to decide if they refer to the same "case item type" under case.info (e.g. if it's editing the status, info (notes, perpetrators, etc))
- * Example paths are like:
- *   [ 'info', 'perpetrators' ] when adding a new perpetrator
- *   [ 'info', 'perpetrators', 0, 'perpetrator', 'phone1' ] when editin an existing perpetrator
- */
-const isPathTarget =
-  (target: string[]) =>
-  (changePath: string[]): boolean =>
-    target.every((value, index) => changePath[index] === value) &&
-    target.length === changePath.length;
-
-const isDescendantPathTarget =
-  (target: string[]) =>
-  (changePath: string[]): boolean =>
-    target.every((value, index) => changePath[index] === value) &&
-    target.length < changePath.length;
-
-const isPathTargetsStatus = isPathTarget(['status']);
-const isPathTargetsInfo = isDescendantPathTarget(['info']);
-
-const isNewKind = <T>(change: Diff<T>): change is DiffNew<T> =>
-  change.kind === NEW_PROPERTY;
-
-const isArrayKind = <T>(change: Diff<T>): change is DiffArray<T> =>
-  change.kind === ARRAY_CHANGED;
-
-const isNewOrAddKind = <T>(change: Diff<T>) => isNewKind(change) || isArrayKind(change);
-
-const isEditKind = <T>(change: Diff<T>): change is DiffEdit<T> =>
-  change.kind === EDITED_PROPERTY;
```

Comment:
Bye bye, see you never 😅

Reply:
Hahaha I also enjoy removing it but have nostalgia of doing so, I remember when you first introduced the diffing algo and it seemed so cool.